flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/table/StateDynamicTableSourceFactory.java

```diff
@@ -0,0 +1,355 @@
+            case ARRAY:
+                return StateConnectorOptions.StateType.LIST;
+
+            case MAP:
+                return StateConnectorOptions.StateType.MAP;
+
+            default:
+                return StateConnectorOptions.StateType.VALUE;
+        }
+    }
+
+    @Nullable
+    private String inferStateMapKeyFormat(String columnName, LogicalType logicalType) {
+        return logicalType.is(LogicalTypeRoot.MAP)
+                ? inferStateValueFormat(columnName, ((MapType) logicalType).getKeyType())
+                : null;
+    }
+
+    private String inferStateValueFormat(String columnName, LogicalType logicalType) {
+        switch (logicalType.getTypeRoot()) {
+            case CHAR:
+            case VARCHAR:
+                return String.class.getName();
+
+            case BOOLEAN:
+                return Boolean.class.getName();
+
+            case BINARY:
+            case VARBINARY:
+                return byte[].class.getName();
+
+            case DECIMAL:
+                return BigDecimal.class.getName();
+
+            case TINYINT:
+                return Byte.class.getName();
+
+            case SMALLINT:
+                return Short.class.getName();
+
+            case INTEGER:
+                return Integer.class.getName();
+
+            case BIGINT:
+                return Long.class.getName();
+
+            case FLOAT:
+                return Float.class.getName();
+
+            case DOUBLE:
+                return Double.class.getName();
+
+            case DATE:
+                return Integer.class.getName();
+
+            case INTERVAL_YEAR_MONTH:
+            case INTERVAL_DAY_TIME:
+                return Long.class.getName();
+
+            case ARRAY:
+                return inferStateValueFormat(
+                        columnName, ((ArrayType) logicalType).getElementType());
+
+            case MAP:
+                return inferStateValueFormat(columnName, ((MapType) logicalType).getValueType());
+
+            case NULL:
+                return null;
+
+            case ROW:
+            case MULTISET:
+            case TIME_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITH_TIME_ZONE:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+            case DISTINCT_TYPE:
+            case STRUCTURED_TYPE:
+            case RAW:
+            case SYMBOL:
+            case UNRESOLVED:
+            default:
+                throw new UnsupportedOperationException(
+                        String.format(
+                                "Unable to infer state format for SQL type: %s in column: %s. "
+                                        + "Please override the type with the following config parameter: %s.%s.%s",
+                                logicalType, columnName, FIELDS, columnName, VALUE_FORMAT));
+        }
+    }
+
+    @Override
+    public String factoryIdentifier() {
+        return "state";
```

Comment:
I think the identifier should be 'savepoint' as the FLIP describes.

Reply:
It would be great if also changing the class name to 'SavepointDynamicTableSourceFactory'

Reply:
Oh yeah, leftover what I've forgotten. Fixing it...